src/Build.OM.UnitTests/Definition/ProjectItem_Tests.cs

```diff
@@ -2713,5 +2715,29 @@ private static ProjectItem GetOneItem(string content)
             Assert.Equal(1, items.Count);
             return items[0];
         }
+
+        /// <summary>
+        /// Item metadata "Filename" should not depends on platform specific slashes.
+        /// </summary>
+        [Fact]
+        public void FileNameMetadataEvaluationShouldNotDependsFromPlatformSpecificSlashes()
+        {
+            using (var env = TestEnvironment.Create())
+            using (var projectCollection = new ProjectCollection())
+            {
+                var testFiles = env.CreateTestProjectWithFiles(@"<?xml version=`1.0` encoding=`utf-8`?>
+                    <Project ToolsVersion=`msbuilddefaulttoolsversion` DefaultTargets=`Validate` xmlns=`msbuildnamespace`>
+                      <ItemGroup>
+                        <A Include=`A\B\C\D.cs` />
+                        <B Include=`@(A->'%(Filename)_test.ext')` />
+                      </ItemGroup>
+                    </Project>");
+                var project = new Project(testFiles.ProjectFile, new Dictionary<string, string>(), null, projectCollection);
+                var buildManager = BuildManager.DefaultBuildManager;
+                var projectInstance = buildManager.GetProjectInstanceForBuild(project);
+                var itemB = projectInstance.Items.Single(i => i.ItemType == "B").EvaluatedInclude.Replace("/", "\\");
```

Comment:
Is this `Replace` needed? There shouldn't be any slashes in the include now, right?